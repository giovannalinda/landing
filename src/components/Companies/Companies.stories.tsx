/* eslint-disable @next/next/no-img-element */
import { Meta, Story } from '@storybook/react'

import { Assets } from '~/assets'

import { Companies } from './Companies'
import * as S from './Companies.styled'

export default {
  title: 'Components/Companies',
  component: Companies,
} as Meta

export const Company: Story = () => (
  <S.CompaniesList>
    <li>
      <S.CompanyCard target='_blank' rel='noreferrer' href='https://google.com'>
        <img src={Assets.LogoGoogle} width='113' height='26' />
      </S.CompanyCard>
    </li>
  </S.CompaniesList>
)
