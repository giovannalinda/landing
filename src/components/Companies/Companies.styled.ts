import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  max-width: ${theme.grid.lg};
  width: 100%;
  margin: 18rem auto 0;
  display: flex;
  justify-content: space-between;

  p {
    font-size: 3.2rem;
    max-width: 31.2rem;

    span {
      color: ${theme.colors.secondaryPink};

      &::before {
        content: '';
        position: absolute;
        width: 8.8rem;
        height: 2rem;
        background: ${theme.colors.secondaryPurple};
        opacity: 0.15;
        z-index: 0;
        margin-top: ${theme.space[4]};
      }
    }
  }
`

export const CompaniesList = styled.ul`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: ${theme.space[5]};
  width: 100%;
  margin-left: 7.2rem;

  li {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 12rem;
    background: ${theme.colors.white};
    border-radius: ${theme.borderRadius.card};
    transition: 0.2s;
    cursor: pointer;

    svg {
      color: ${theme.colors.secondaryPurple};
    }

    &:hover {
      transform: translateY(-0.5rem);
    }
  }
`
