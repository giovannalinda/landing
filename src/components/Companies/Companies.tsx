import Image from 'next/image'
import { CgArrowLongRight } from 'react-icons/cg'

import { Assets } from '~/assets'
import { useTranslation } from '~/lib/useTranslation'

import * as S from './Companies.styled'

const companies = [
  {
    name: 'YouTube',
    logo: Assets.LogoYoutube,
    url: 'https://youtube.com',
  },
  {
    name: 'Google',
    logo: Assets.LogoGoogle,
    url: 'https://google.com',
  },
  {
    name: 'Adobe',
    logo: Assets.LogoAdobe,
    url: 'https://adobe.com',
  },
  {
    name: 'Sketch',
    logo: Assets.LogoSketch,
    url: 'https://sketch.com',
  },
]

export function Companies() {
  const { t } = useTranslation()

  return (
    <S.Container>
      <h1>
        {`${t('I worked with')} `}
        <span>289+</span> {t('Companies all over the World.')}
      </h1>

      <S.CompaniesList>
        {companies.map((company) => (
          <li key={company.name} aria-label={company.name}>
            <S.CompanyCard target='_blank' rel='noreferrer' href={company.url}>
              <Image src={company.logo} width='113' height='26' />
            </S.CompanyCard>
          </li>
        ))}

        <S.CompanyCard as='li'>
          <CgArrowLongRight size={30} />
        </S.CompanyCard>
      </S.CompaniesList>
    </S.Container>
  )
}
