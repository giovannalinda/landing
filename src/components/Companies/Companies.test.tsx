import { render } from '~/__helpers__/app-tests'
import { Companies } from './Companies'

describe('Companies', () => {
  it('should match snapshot', () => {
    const { container } = render(<Companies />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
