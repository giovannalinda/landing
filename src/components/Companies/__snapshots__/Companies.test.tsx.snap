// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies should match snapshot 1`] = `
<section
  class="sc-bczRLJ kVzqFo"
>
  <h1>
    I worked with 
    <span>
      289+
    </span>
     
    Companies all over the World.
  </h1>
  <ul
    class="sc-gsnTZi gmoSjh"
  >
    <li
      aria-label="YouTube"
    >
      <a
        class="sc-dkzDqf emVUNr"
        href="https://youtube.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          height="26"
          src="/youtubelogo.svg"
          width="113"
        />
      </a>
    </li>
    <li
      aria-label="Google"
    >
      <a
        class="sc-dkzDqf emVUNr"
        href="https://google.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          height="26"
          src="/googlelogo.svg"
          width="113"
        />
      </a>
    </li>
    <li
      aria-label="Adobe"
    >
      <a
        class="sc-dkzDqf emVUNr"
        href="https://adobe.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          height="26"
          src="/adobelogo.svg"
          width="113"
        />
      </a>
    </li>
    <li
      aria-label="Sketch"
    >
      <a
        class="sc-dkzDqf emVUNr"
        href="https://sketch.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          height="26"
          src="/sketchlogo.svg"
          width="113"
        />
      </a>
    </li>
    <li
      class="sc-dkzDqf emVUNr"
    >
      <svg
        fill="none"
        height="30"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
          fill="currentColor"
        />
      </svg>
    </li>
  </ul>
</section>
`;
